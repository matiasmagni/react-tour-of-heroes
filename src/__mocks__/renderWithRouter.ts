import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

const renderWithRouter = (ui: ReactElement, route: string = '/') => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
};

export default renderWithRouter;
