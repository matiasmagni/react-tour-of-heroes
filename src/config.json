{
    "API_URL": {
        "heroes": "http://localhost:5000/hero" 
    }
}
