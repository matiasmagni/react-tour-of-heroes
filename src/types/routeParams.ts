export interface IRouteParams {
  heroId: string;
}
