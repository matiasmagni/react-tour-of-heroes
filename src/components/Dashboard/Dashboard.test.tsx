import { screen, waitFor } from '@testing-library/react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import renderWithRouter from '../../__mocks__/renderWithRouter';
import Dashboard from './Dashboard';
import config from '../../config.json';
import heroesMock from '../../__mocks__/heroesMock';
import { IHero } from '../../types/hero';

const mockAxios = new MockAdapter(axios);

describe('Tests del componente Dashboard.', () => {
  it('El componente Dashboard se renderiza correctamente.', async () => {
    const spyAxiosGet = jest.spyOn(axios, 'get');
    const tree = renderWithRouter(<Dashboard />, '/dashboard');
    mockAxios.onGet(config.API_URL.heroes).reply(200, heroesMock);

    await waitFor(() => {
      heroesMock.forEach((hero: IHero) => {
        expect(screen.getByText(hero.name!)).toBeInTheDocument();
      });
    });

    expect(spyAxiosGet).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
