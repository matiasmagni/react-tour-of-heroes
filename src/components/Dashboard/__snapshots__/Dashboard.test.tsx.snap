// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests del componente Dashboard. El componente Dashboard se renderiza correctamente. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2>
        Top Heroes
      </h2>
      <div
        class="heroes-menu"
      >
        <a
          href="/hero-detail/12"
        >
          Pepito
        </a>
        <a
          href="/hero-detail/13"
        >
          Bombasto
        </a>
        <a
          href="/hero-detail/14"
        >
          Celeritas
        </a>
        <a
          href="/hero-detail/15"
        >
          Magneta
        </a>
        <a
          href="/hero-detail/16"
        >
          RubberMan
        </a>
        <a
          href="/hero-detail/17"
        >
          Dynama
        </a>
        <a
          href="/hero-detail/18"
        >
          Dr IQ
        </a>
        <a
          href="/hero-detail/19"
        >
          Magma
        </a>
        <a
          href="/hero-detail/20"
        >
          Tornado
        </a>
      </div>
      <label
        for="search"
      >
        Hero Search:
        <input
          id="search"
          name="search"
          placeholder="Hero name"
          type="text"
          value=""
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <h2>
      Top Heroes
    </h2>
    <div
      class="heroes-menu"
    >
      <a
        href="/hero-detail/12"
      >
        Pepito
      </a>
      <a
        href="/hero-detail/13"
      >
        Bombasto
      </a>
      <a
        href="/hero-detail/14"
      >
        Celeritas
      </a>
      <a
        href="/hero-detail/15"
      >
        Magneta
      </a>
      <a
        href="/hero-detail/16"
      >
        RubberMan
      </a>
      <a
        href="/hero-detail/17"
      >
        Dynama
      </a>
      <a
        href="/hero-detail/18"
      >
        Dr IQ
      </a>
      <a
        href="/hero-detail/19"
      >
        Magma
      </a>
      <a
        href="/hero-detail/20"
      >
        Tornado
      </a>
    </div>
    <label
      for="search"
    >
      Hero Search:
      <input
        id="search"
        name="search"
        placeholder="Hero name"
        type="text"
        value=""
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
